File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    /**
     * 查询所有
     *
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }



    /**
     * 分页查询所有
     * 条件查询
     * @param brand
     * @param page
     * @param size
     * @RequestBody 该注释表示请求方式为post
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(@RequestBody TbBrand brand, int page, int size) {
        return brandService.findPage(brand,page, size);
    }

    /**
     * 新建
     *
     * @param brand
     * @return
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            return new Result(true, "新建成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "创建失败");
        }
    }

    /**
     * 删除
     * @param ids
     * @return
     */
    @RequestMapping("/del")
    public Result del(Long[] ids){
       try {
           brandService.del(ids);
           return new Result(true,"删除成功");
       }catch (Exception e){
           e.printStackTrace();
           return new Result(false,"删除失败");
       }
    }

    /**
     * 查找单条数据
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    /**
     * 修改
     * @param brand
     * @return
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    /**
     * 添加下拉框
     * @return
     */
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function ($scope,$controller, brandService) {
    $controller('baseController',{$scope:$scope});//继承
    //查询品牌列表
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        );
    };

    //分页
    /*  $scope.findPage = function (page, size) {
          $http.post('../brand/findPage.do?page=' + page + '&size=' + size).success(
              function (response) {
                  $scope.list = response.rows;//显示当前页数据
                  $scope.paginationConf.totalItems = response.total;//更新总记录数
              }
          );
      };*/

    //新建与修改
    $scope.save = function () {
        var serviceObject;
        if($scope.entity.id != null){
            serviceObject = brandService.update($scope.entity);
        }else {
            serviceObject = brandService.add($scope.entity);
        }
        serviceObject.success(
            function (response) {
                if (response.success) {
                    //如果保存成功则重新加载
                    alert(response.message);
                    $scope.reloadList();
                } else {
                    alert(response.message)
                }
            });
    };
    //查找单条数据进行修改
    $scope.findOne=function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity=response;
            }
        );
    };

    //批量删除
    $scope.del = function () {
        if (confirm("确定要删除吗")) {
            brandService.del($scope.selectIds).success(
                function (response) {
                    if (response.success) {
                        $scope.reloadList();
                    } else {
                        alert(response.message)
                    }
                }
            );
        }
    };

    //条件查询
    $scope.searchEntity={};

    $scope.search = function (page, size) {
        brandService.search(page,size,$scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;//显示当前页数据
                $scope.paginationConf.totalItems = response.total;//更新总记录数
            }
        );
    }


});